lib/context/form_context.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useContext, FC, ReactNode } from 'react';
+import useSWRMutation from 'swr/mutation';
+
+interface Form {
+  latLong: [number, number];
+  setFormData: (direction: number, tilt: number, capacity: number) => void;
+  setMapData: (lat: number, long: number) => void;
+}
+
+interface FormProviderProps {
+  children: ReactNode | ReactNode[];
+}
+
+type FormPostData = {
+  site_uuid: number;
+  client_name: string;
+  client_site_id: number;
+  client_site_name: string;
+  latitude: number;
+  longitude: number;
+  installed_capacity_kw: number;
+  created_utc: string;
+  updated_utc: string;
+  orientation: number;
+  tilt: number;
+};
+
+async function sendRequest(url: string, { arg }: { arg: FormPostData }) {
+  return fetch(url, {
+    method: 'POST',
+    body: JSON.stringify(arg),
+    headers: {
+      'content-type': 'application/json',
+    },
+  }).then((res) => res.json());
+}
+
+const FormContext = React.createContext<Form | null>(null);
+
+const FormProvider: FC<FormProviderProps> = ({ children }) => {
```

Comment:
Instead of manually adding a type for the `children` prop, React provides one you can use in `PropsWithChildren`. You'll need to import this though.
```suggestion
const FormProvider: FC<PropsWithChildren> = ({ children }) => {
```